api/agent.go

```diff
@@ -641,6 +641,7 @@ func (a *Agent) AgentHealthServiceByNameOpts(service string, q *QueryOptions) (s
 	r.setQueryOptions(q)
 	r.params.Add("format", "json")
 	r.header.Set("Accept", "application/json")
+	// not a lot of value in wrapping the doRequest call in a requireHttpCodes call
```

Comment:
Why's that?

Reply:
tried to answer and consolidate the convo here --> https://github.com/hashicorp/consul/pull/11158/files#r717131243